src/tests/Loader/classloader/InlineArray/InlineArrayInvalid.cs

```diff
@@ -0,0 +1,113 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using Xunit;
+
+unsafe class Validate
+{
+    [InlineArray(1)]
+    [StructLayout(LayoutKind.Explicit)]
+    private struct Explicit
+    {
+        [FieldOffset(0)]
+        public Guid Guid;
+    }
+
+    [Fact]
+    public static void Explicit_Fails()
+    {
+        Console.WriteLine($"{nameof(Explicit_Fails)}...");
+        Assert.Throws<TypeLoadException>(() => { var t = typeof(Explicit); });
+
+        Assert.Throws<TypeLoadException>(() =>
+        {
+            return sizeof(Explicit);
+        });
+    }
+
+    [InlineArray(0)]
+    private struct ZeroLength
+    {
+        public int field;
+    }
+
+    [Fact]
+    public static void ZeroLength_Fails()
+    {
+        Console.WriteLine($"{nameof(ZeroLength_Fails)}...");
+        Assert.Throws<TypeLoadException>(() => { var t = typeof(ZeroLength); });
+
+        Assert.Throws<TypeLoadException>(() =>
+        {
+            var t = new ZeroLength()
+            {
+                field = 1
+            };
+            return t;
+        });
+    }
+
+    [InlineArray(0x20000000)]
+    private struct TooLarge
+    {
+        public long field;
+    }
```

Comment:
We should also have a test for negative numbers.

Reply:
good idea!